### DISCUSSION TITLE
Not getting updated values in table columns

### DISCUSSION THREAD

**Discussion by Nav-Chouhan** [Laravel-Backpack/community-forum#445] [Q&A (Help)] [answered]
I am having a customer table at which there are 3 columns **city, state, and country**. Unfortunately, when we create a new record or update a record these 3 columns are not having their values init.

I am using the select_from_array custom field for **state** and **city** fields. Basically, state and city fields are dependent select box. When Country gets selected an ajax call trigger to get the state records and populate the state dropdown. The same is happening for the city dropdown it gets its value when the state gets selected.

For more details please have a look at the attached images:
<img width="774" alt="Screenshot 2023-03-18 at 2 42 12 PM" src="https://user-images.githubusercontent.com/769561/226098154-9bf8bcf4-53d1-4b17-b054-88265aa8df58.png">

<img width="679" alt="Screenshot 2023-03-18 at 3 17 53 PM" src="https://user-images.githubusercontent.com/769561/226098156-142109ee-462f-4f12-a0f6-ce3864330ec6.png">

<img width="495" alt="Screenshot 2023-03-18 at 3 18 17 PM" src="https://user-images.githubusercontent.com/769561/226098157-a8ce9a43-fdfc-4d13-bae6-6d8a53a9c0c2.png">

Also, Would like to update you that when I debug it and check the request parament after the form post I found the values are appearing properly.

<img width="177" alt="Screenshot 2023-03-18 at 3 23 26 PM" src="https://user-images.githubusercontent.com/769561/226098722-9f4a341e-86ff-4820-9fcf-55cd03075626.png">

And after the update command 

<img width="412" alt="Screenshot 2023-03-18 at 3 25 29 PM" src="https://user-images.githubusercontent.com/769561/226098746-d13fac89-4e9d-47b0-8931-3de740ea9366.png">

Please guide me on what exactly happening here.

Thanks In Advance

**Comment by promatik** [comment, marked as answer]
Hi @Nav-Chouhan!

I see many issues in your code;

1) Models should be in singular form, so Costumers » Costumer, Cities » City, States » State, Countries » Country.
2) When you have relations, you should use `_id`, so `city` » `city_id`, `state` » `state_id`, `country` » `country_id`.

Internally, some of this things would make the relations to be saved automatically, you need extra setup if you don't follow the standards.

I think the issue may be in the definition of the belongsTo relations, since you're not following the standard, you must set the column name there `$this->belongsTo(Categories::class, 'category');`

BUT, it's highly recommended to follow the standards, instead of fixing this inconsistencies every time they happen, otherwise you'll keep on having more problems in the future.

**Comment by Nav-Chouhan** [comment]
@promatik Thanks for highlighting the bugs. I have changed the code according to the standards. Now of fine.

Thanks Again :)
